rsfbclient-native/src/params.rs

```diff
@@ -123,9 +137,66 @@ impl ParamBuffer {
         var.sqldata = buffer.as_mut_ptr() as *mut _;
         var.sqllen = buffer.len() as i16;
 
-        ParamBuffer {
+        Ok(ParamBuffer {
             _buffer: buffer.into_boxed_slice(),
             _nullind: nullind,
+        })
+    }
+}
+
+// Convert the binary vec to a blob
+fn binary_to_blob(
+    buffer: Vec<u8>,
+    db_handle: &mut ibase::isc_db_handle,
+    tr_handle: &mut ibase::isc_tr_handle,
+    ibase: &IBase,
+) -> Result<Vec<u8>, FbError> {
+    let mut status = Status::default();
+    let mut handle = 0;
+
+    let mut blob_id = MaybeUninit::<ibase::GDS_QUAD_t>::uninit();
+
+    unsafe {
+        if ibase.isc_create_blob()(
+            &mut status[0],
+            db_handle,
+            tr_handle,
+            &mut handle,
+            blob_id.as_mut_ptr(),
+        ) != 0
+        {
+            return Err(status.as_error(&ibase));
+        }
+    }
+
+    // Assert that the handle is valid
+    debug_assert_ne!(handle, 0);
+
+    let buffer_i8: Vec<i8> = buffer.iter().map(|byte| *byte as i8).collect();
+
+    unsafe {
+        if ibase.isc_put_segment()(
+            &mut status[0],
+            &mut handle,
+            buffer_i8.len() as u16,
+            buffer_i8.as_ptr(),
```

Comment:
The `buffer_i8` is not needed.
```suggestion
    unsafe {
        if ibase.isc_put_segment()(
            &mut status[0],
            &mut handle,
            buffer.len() as u16,
            buffer.as_ptr() as *mut i8,
```

Reply:
Done